boundary/collect.py

```diff
@@ -104,12 +104,8 @@ def process_passes(self, p, _):
         self.final_work()
 
     def register_cbs(self):
-        if hasattr(self, 'var_declare'):
-            gcc.register_callback(gcc.PLUGIN_FINISH_DECL, self.var_declare)
         if hasattr(self, 'final_work'):
```

Comment:
This can be removed too, because we know for certain there's a `final_work` method now. Just `register_callback` in \_\_main\_\_.